src/sandstorm/web-session.capnp

```diff
@@ -97,6 +107,32 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
 
     accept @2 :List(AcceptedType);
     # This corresponds to the Accept header
+
+    additionalHeaders @3 :List(Header);
+    # Additional headers present in the request. Only whitelisted headers are
+    # permitted.
+
+    struct Header {
+      name @0 :Text;  # lower-cased name
+      value @1 :Text;
+    }
+
+    const headerWhitelist :List(Text) = [
+      "if-match",              # webdav etag
```

Comment:
`If-Match` and `If-Not-Match` are standard HTTP headers which we really need to support explicitly rather than pass through. Passing in these headers will affect non-webdav apps too, and cause them to possibly return new response codes that they didn't before, which we may not be prepared to handle. So, I don't think they belong on the whitelist, which is intended to be for non-standard things.

I think arguably we should add this to `Context`:

    union etagPrecondition {
      none @4 :Void;  # No precondition.
      exists @5 :Void;  # If-Match: *
      matchesOneOf @6 :List(Text);  # If-Match
      matchesNoneOf @7 :List(Text);  # If-None-Match
    }